### PR TITLE
fsevent-sys 3.0.0

### PR DESCRIPTION
**Pull request by mtak-**
Update `fsevent-sys` to 3.0.0.

Related: #211

### CODE REVIEW

**Comment by 0xpr03** [comment]
So we have to switch to 3.X ? Or can we just stay at 2 as you yanked the higher version ?

**Comment by Dylan-DPC** [comment]
@0xpr03 They yanked 2.1 i think not 3.0 . this looks good to me. Do you have any other concerns (other than tests failing which could be unrelated)

**Comment by 0xpr03** [comment]
If we rebase this on the current try-v4 branch and thus also retrigger travis, we'd have a working 1.26 CI check now.

**Comment by 0xpr03** [comment]
I'll add a correct Cargo.lock.min for the rust 1.26 checks later, after rebasing on the current try-v4 and with that also re-trigger travis. If that works I'll merge it.

**Comment by 0xpr03** [comment]
@mtak- your changes break mac builds completely
see [here](https://travis-ci.com/notify-rs/notify/jobs/237900612#L146)

**Comment by 0xpr03** [comment]
It seems like mac builds are failing for 1.26 and tests fail for stable & nightly.

**Comment by mtak-** [comment]
Upstream PR to fsevent-sys to fix 1.26.1 compatibility: https://github.com/octplane/fsevent-rust/pull/23
Pushed up some 1.26.1 fixes. Waiting to see what the tests say now.

**Comment by mtak-** [comment]
I don't see any failures relevant to this PR.

I would recommend merging it as pre `3.0` `fsevents-sys` had some UB in it's ffi.

**Comment by 0xpr03** [comment]
Please do not fix filetime to 0.2.5, we do this via Cargo.lock.min files during CI.

File: src/fsevent.rs [hunk 1]
```diff
@@ -255,13 +255,35 @@ impl FsEventWatcher {
...
-    info: *mut libc::c_void,
-    num_events: libc::size_t,                // size_t numEvents
-    event_paths: *const *const libc::c_char, // void *eventPaths
-    event_flags: *mut libc::c_void,          // const FSEventStreamEventFlags eventFlags[]
-    event_ids: *mut libc::c_void,            // const FSEventStreamEventId eventIds[]
+    info: *mut ::std::os::raw::c_void,
+    num_events: libc::size_t,                   // size_t numEvents
+    event_paths: *mut ::std::os::raw::c_void,   // void *eventPaths
+    event_flags: *const ::std::os::raw::c_void, // const FSEventStreamEventFlags eventFlags[]
+    event_ids: *const ::std::os::raw::c_void,   // const FSEventStreamEventId eventIds[]
```
Comment (JohnTitor) [inline src/fsevent.rs]: It seems the signature is changed, it's fine?

File: Cargo.toml [hunk 2]
```diff
@@ -33,8 +33,8 @@ mio = "^0.6.15"
 mio-extras = "^2.0.5"
 
 [target.'cfg(target_os="macos")'.dependencies]
-fsevent = "0.4"
-fsevent-sys = "2"
+fsevent = { git = "https://github.com/mtak-/fsevent-rust", branch = "downstream-test" }
+fsevent-sys = { git = "https://github.com/mtak-/fsevent-rust", branch = "downstream-test" }
```
Comment (JohnTitor) [inline Cargo.toml]: Hm, the pull request hasn't been reviewed yet. Should we ping the fsevent's author, or fork into notify-rs and use it?

**Comment by erickt** [comment]
Should this be closed? fsevent-sys is currently at 3.0.0.

**Pull request closed without merging**